src/lib/authist.ts

```diff
@@ -2,15 +2,18 @@ import { NextFunction, Request, Response } from 'express';
 import { IncomingMessage } from 'http';
 import { expressBearer } from './authMiddleware';
 import { EmailPasswordProviderOptions, signInWithEmailAndPassword } from './providers/emailPasswordProvider';
+import { InstagramProviderOptions, signInWithInstagram } from './providers/instagramProvider';
 
 export const createAuthenticator = (options: AuthistOptions): Authenticator => ({
     expressBearer: expressBearer(options),
     signInWithEmailAndPassword: signInWithEmailAndPassword(options),
+    signInWithInstagram: signInWithInstagram(options),
 });
 
 export interface Authenticator {
     expressBearer: (request: Request, response: Response, next: NextFunction) => Promise<any>;
     signInWithEmailAndPassword: (email: string, password: string, req?: IncomingMessage) => Promise<UserCredentials>;
+    signInWithInstagram: (token: string, req?: IncomingMessage) => Promise<UserCredentials>;
```

Comment:
What is this token? Please add a comment or readme docs. Not sure for instagram, but there can be several tokens (some acting like access tokens, some just only as a auth code you use in order to exchange it for the access token)

Reply:
also have you considered using types from the provider directly? `ReturnType<typeof signInWithInstagram>` if its possible.. so you won't have to write it by hand :)

Reply:
yes it's the facebook access token (any of the page / user / app / long lived tokens) I added links to the readme

I used the `ReturnType` it works fine! Thanks

Reply:
I think I will merge some functionality with facebook provider later - it's the same api but the `fields` and `scopes` are different. There are also some differences between user token and page token